### PR TITLE
Refactor daily CI configuration using matrix

### PR DESCRIPTION
**Pull request by PragmaTwice**
- solve https://github.com/apache/incubator-kvrocks/pull/626#issuecomment-1154655734 found by @ShooterIT 
- add pull_request trigger proposed by @tisonkun in https://github.com/apache/incubator-kvrocks/pull/626#issuecomment-1154664953
- refactor the daily CI using a unified matrix instead of some repeated jobs

### CODE REVIEW

**Comment by PragmaTwice** [comment]
The daily CI will probably always fail due to the error reported by TSan, so I'm wondering if I should wait for the error to be fixed before merging in (block this PR temporarily), or merge this PR in first and then consider fixing the error? Thanks~ @git-hulk

**Comment by git-hulk** [comment]
@PragmaTwice How about removing or disabling the TSAN condition first before we fixed those data races? I'll follow up those data race issues recently.

**Comment by PragmaTwice** [comment]
> @PragmaTwice How about removing or disabling the TSAN condition first before we fixed those data races? I'll follow up those data race issues recently.

Sounds great. I will remove them temporarily.

**Review by git-hulk — APPROVED** [review]

**Comment by git-hulk** [comment]
Wow, thanks for your great work,

File: .github/workflows/daily-ci.yaml
Comment (tisonkun) [inline .github/workflows/daily-ci.yaml:72]: Why we comment out these lines?

Comment (PragmaTwice) [reply .github/workflows/daily-ci.yaml:72]: Refer to https://github.com/apache/incubator-kvrocks/pull/630#issuecomment-1158446352, we temporarily remove these job since it will fail due to some data race in kvrocks, and it will be re-add after the bug is fixed.

Comment (tisonkun) [reply .github/workflows/daily-ci.yaml:72]: OK. Then we should open an issue once this PR merged.

**Review by tisonkun — APPROVED** [review]

**Comment by PragmaTwice** [comment]
These removed jobs will be re-added while https://github.com/apache/incubator-kvrocks/issues/639 is solved.

**Comment by PragmaTwice** [comment]
Thanks everyone, merging...

**Pull request merged**
